githublib.py

```diff
@@ -0,0 +1,115 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2021 Red Hat, Inc.
+#
+
+import time
+import requests
+
+from datetime import datetime
+
+from config import Config
+
+SECONDS_PER_WEEK = 7 * 24 * 60 * 60
+# max github pages to process
+GITHUB_MAX_PAGES = 20
+# process upto x weeks back
+MAX_DELTA_WEEKS = 4
+_ENV_VAR_NAMES = ["GITHUB_TOKEN", "GITHUB_OWNER", "GITHUB_REPO", "GITHUB_LABEL"]
+
+
+def config():
+    c = Config(_ENV_VAR_NAMES)
+    c.Load()
+    return c
+
+
+class Issue:
+    def __init__(self, issue):
+        self.issue = issue
+
+    def filter(self, expected_label):
+        if expected_label == "":
+            return True
+        for label in self.issue["labels"]:
+            if label["name"] == expected_label:
+                return True
+        return False
+
+    @property
+    def repo(self):
+        return self.issue["html_url"].split("/")[4]
+
+    @property
+    def id(self):
+        return self.issue["number"]
+
+    @property
+    def url(self):
+        return self.issue["html_url"]
+
+    @property
+    def title(self):
+        return self.issue["title"]
+
+    @property
+    def epoch(self):
+        TIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+        timestamp = self.issue["created_at"]
+        return int(datetime.strptime(timestamp, TIME_FORMAT).timestamp())
+
+
+class Github:
+    def __init__(self, cfg):
+        owner = cfg.vars["GITHUB_OWNER"]
+        repo = cfg.vars["GITHUB_REPO"]
+        self.query_url = f"https://api.github.com/repos/{owner}/{repo}/issues"
+        self.headers = {"Authorization": f"token {cfg.vars['GITHUB_TOKEN']}"}
+        self.expected_label = cfg.vars["GITHUB_LABEL"]
+
+    def issues(self):
+        for page in range(1, GITHUB_MAX_PAGES):
+            params = {"state": "open", "page": page, "per_page": "100"}
+            r = requests.get(self.query_url, headers=self.headers, params=params)
+            issues = r.json()
+
+            if len(issues) == 0:
+                return
+
+            yield issues
+
+    def filter(self, issues):
+        for element in issues:
+            issue = Issue(element)
+            if "pull" in issue.url:
+                continue
+
+            if issue.filter(self.expected_label) and filter_by_time(
+                issue, MAX_DELTA_WEEKS
+            ):
+                yield (issue)
+
+    def get_issue(self, issue_id):
+        r = requests.get(f"{self.query_url}/{issue_id}", headers=self.headers)
+        issue = r.json()
+        if issue.get("url", None) is None:
+            return None
+        return Issue(issue)
+
+
+def filter_by_time(issue, max_delta_weeks):
```

Comment:
"filter" is usually associated to a collection, but you seem to check a specific condition for a specific item.
The name should reflect that you return a boolean in such a case.

You could make it a collection, by passing the collection and returning the filtered one back.